tailscale/resource_dns_split_nameservers.go

```diff
@@ -6,24 +6,25 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
-	"github.com/tailscale/tailscale-client-go/tailscale"
+	tsclient "github.com/tailscale/tailscale-client-go/v2"
 )
 
 func resourceDNSSplitNameservers() *schema.Resource {
 	return &schema.Resource{
 		Description:   "The dns_split_nameservers resource allows you to configure split DNS nameservers for your Tailscale network. See https://tailscale.com/kb/1054/dns for more information.",
 		ReadContext:   resourceSplitDNSNameserversRead,
-		CreateContext: resourceSplitDNSNameserversCreate,
-		UpdateContext: resourceSplitDNSNameserversUpdate,
+		CreateContext: resourceSplitDNSNameserversCreateOrUpdate,
+		UpdateContext: resourceSplitDNSNameserversCreateOrUpdate,
 		DeleteContext: resourceSplitDNSNameserversDelete,
 		Importer: &schema.ResourceImporter{
 			StateContext: schema.ImportStatePassthroughContext,
 		},
 		Schema: map[string]*schema.Schema{
 			"domain": {
 				Type:        schema.TypeString,
-				Description: "Domain to configure split DNS for. Requests for this domain will be resolved using the provided nameservers.",
+				Description: "Domain to configure split DNS for. Requests for this domain will be resolved using the provided nameservers. This cannot be changed after the resource has been configured.",
```

Comment:
nit: I can see what this update is trying to get at, but it makes it sound a bit like a permanent tailnet setting. I think the documentation is still ok without an update because users will get to see the "force new" behaviour in their plan output, but if you want to make it more explicit, how about something similar to "Updating this field will force the resource to be recreated"